File: src/jskills/elo/GaussianKFactor.java
using System;

namespace Moserware.Skills.Elo
{
    public class GaussianKFactor : KFactor
    {
        // From paper
        final double StableDynamicsKFactor = 24.0;

        public GaussianKFactor()
            : base(StableDynamicsKFactor)
        {
        }

        public GaussianKFactor(GameInfo gameInfo, double latestGameWeightingFactor)
            : base(latestGameWeightingFactor * gameInfo.Beta * Math.Sqrt(Math.PI))
        {
        }
    }
}

File: src/jskills/elo/GaussianKFactor.scala
package jskills.elo;

import jskills.GameInfo;

class GaussianKFactor(value: Double = GaussianKFactor.StableDynamicsKFactor) extends KFactor(value) {
  def this(gameInfo: GameInfo, latestGameWeightingFactor: Double) {
    this(latestGameWeightingFactor * gameInfo.getBeta() * Math.sqrt(Math.Pi))
  }
}
object GaussianKFactor {
  // From paper
  val StableDynamicsKFactor = 24.0;
}